rkovalenko/CircleOfChainedWords.cs

```diff
@@ -0,0 +1,89 @@
+﻿namespace TestConsoleApp
+{
+    public static class CircleOfChainedWords
+    {
+        public static bool IsLooped(List<string> listOfWords)
+        {
+            if(!listOfWords.Any() || listOfWords.Count == 1)
+            {
+                return false;
+            }
+
+            var startingWordIndex = 0;
+            var count = listOfWords.Count;
+            for(var i = 1; i < count;)
+            {
+                // compare last char of one word and the first char of the second word
+                if (listOfWords[startingWordIndex][^1] == listOfWords[i][0])
+                {
+                    if(i - startingWordIndex > 1)
+                    {
+                        var overlappingIndex = (startingWordIndex + 1) % count;
+                        (listOfWords[i], listOfWords[overlappingIndex]) = (listOfWords[overlappingIndex], listOfWords[i]);
+                    }
+
+                    ++startingWordIndex;
+
+                    i = startingWordIndex + 1;
+                }
+                else
+                {
+                    ++i;
+                }
+            }
+
+            for(var i = 0; i < count; ++i)
```

Comment:
Not sure why the second loop to double check if we have a full chain from the first one.